pkg/aws/iam_client.go

```diff
@@ -0,0 +1,120 @@
+package aws
+
+import (
+	"context"
+	"errors"
+	"log"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/smithy-go"
+	"github.com/aws/smithy-go/middleware"
+
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/aws/aws-sdk-go-v2/service/iam/types"
+)
+
+type IamWrapper struct {
```

Comment:
I'd suggest creating an interface for this component and having the reconciler depend on the interface instead of the concrete type. It's a good practice and can help when, for example, writing tests, as we can mock that interface in order to test the components that depend on it

Reply:
Yes we did talk about doing an Interface for this.